wasm/tests/data/blinky.rs

```diff
@@ -0,0 +1,70 @@
+#![no_std]
+
+extern crate rustmatic_iec_std as iec_std;
+
+use core::time::Duration;
+use iec_std::intrinsics::{self, wasm_result_t_WASM_SUCCESS as WASM_SUCCESS};
+
+const INPUT_ADDRESS: usize = 0;
+const OUTPUT_ADDRESS: usize = 0;
+const TOGGLED_BIT: u8 = 0;
+
+static mut NEXT_TOGGLE: Duration = Duration::from_secs(0);
+static mut CURRENT_STATE: bool = false;
+
+#[no_mangle]
+pub extern "C" fn poll() {
+    unsafe {
+        let mut secs = 0;
+        let mut nanos = 0;
+        let ret = intrinsics::wasm_current_time(&mut secs, &mut nanos);
+        assert_eq!(ret, WASM_SUCCESS);
+
+        let now = Duration::new(secs, nanos);
+
+        if now > NEXT_TOGGLE {
+            // figure out we need to toggle again
+            NEXT_TOGGLE += polling_period(INPUT_ADDRESS);
+            // and make sure we toggle
+            CURRENT_STATE = !CURRENT_STATE;
+        }
+
+        set_bit_state(OUTPUT_ADDRESS, TOGGLED_BIT, CURRENT_STATE);
+    }
+}
+
+unsafe fn set_bit_state(address: usize, bit: u8, state: bool) {
```

Comment:
Not a blocker for this PR, but do you have any thoughts about how we could wrap these unsafe fn's in safe api's in the future?

Reply:
I'd create a set of functions for reading/writing values of different types. For example,

```rust
fn read_bit(address: usize, bit_offset: u8) -> Result<bool, ReadError> { ... }
fn read_byte(address: usize) -> Result<u8, ReadError> { ... }
fn read_word(address: usize) -> Result<u32, ReadError> { ... }
fn read_double(address: usize) -> Result<f64, ReadError> { ... }

fn write_bit(address: usize, bit_offset: u8, value: bool) -> Result<(), WriteError> { ... }
...
```

I guess `set_bit_state()` doesn't *technically* need to be `unsafe`. The linker will make sure only possible to use `wasm_write_output()` when we're being executed by the rustmatic runtime, and there aren't really any invariants which we can break, so it *should* be safe...